### PR TITLE
[vcpkg-tool] Auto update sha512

### PR DESCRIPTION
**Pull request by autoantwort**
Needed for https://github.com/microsoft/vcpkg-tool/pull/166

### CODE REVIEW

File: scripts/cmake/vcpkg_download_distfile.cmake [hunk 1]
```diff
@@ -70,12 +70,22 @@ function(z_vcpkg_download_distfile_test_hash path kind error_advice sha512 skip_
 
     file(SHA512 "${path}" file_hash)
     if(NOT "${file_hash}" STREQUAL "${sha512}")
-        message(FATAL_ERROR
-            "\nFile does not have expected hash:\n"
+        if(_VCPKG_SHA512_MISMATCH_NO_ERROR)
```
Comment (strega-nil-ms) [inline scripts/cmake/vcpkg_download_distfile.cmake:73]: This should be `Z_VCPKG_SHA512_MISMATCH_NO_ERROR`.

**Review by strega-nil-ms — CHANGES_REQUESTED** [review]

**Comment by PhoebeHui** [comment]
@strega-nil-ms, could you please take a look again?

**Comment by wrobelda** [comment]
Is this still planned for inclusion? It saves tremendous amount of time when mass-updating ports, like I do for KF5 frameworks.

**Comment by JackBoosY** [comment]
Closing this PR since it seems that no progress is being made. Please ping me to reopen if work is still being done.

**Comment by JackBoosY** [comment]
Needs to wait for https://github.com/microsoft/vcpkg-tool/pull/166 merge first.

**Pull request closed without merging**